src/etc/gdb_providers.py

```diff
@@ -0,0 +1,406 @@
+from sys import version_info
+
+import gdb
+from gdb import lookup_type
+
+if version_info[0] >= 3:
+    xrange = range
+
+ZERO_FIELD = "__0"
+FIRST_FIELD = "__1"
+
+
+def unwrap_unique_or_non_null(unique_or_nonnull):
+    # BACKCOMPAT: rust 1.32
+    # https://github.com/rust-lang/rust/commit/7a0911528058e87d22ea305695f4047572c5e067
+    ptr = unique_or_nonnull["pointer"]
+    return ptr if ptr.type.code == gdb.TYPE_CODE_PTR else ptr[ZERO_FIELD]
+
+
+class StructProvider:
```

Comment:
What is the benefit of `StructProvider`?  It seems to me that it's identical to simply letting gdb handle the type itself.

Reply:
Looks like it is needed only for very old GDB or Rust. There was a similar provider in old pretty-printers. So I think I can get rid of it now, thank you.